qiskit/providers/backend_compat.py

```diff
@@ -13,185 +13,275 @@
-                    continue
-                if not properties.is_gate_operational(name, gate.qubits):
-                    continue
+            if name not in all_instructions:
+                logger.info(
+                    "Gate property for instruction %s on qubits %s is found "
+                    "in the BackendProperties payload. However, this gate is not included in the "
+                    "basis_gates or supported_instructions, or maybe the gate model "
+                    "is not defined in the Qiskit namespace. This gate is ignored.",
+                    name,
+                    qubits,
+                )
+                continue
 
-            gate_props = {}
+            in_param = {}
             for param in gate.parameters:
                 if param.name == "gate_error":
-                    gate_props["error"] = param.value
-                if param.name == "gate_length":
-                    gate_props["duration"] = apply_prefix(param.value, param.unit)
-            gates[name][qubits] = InstructionProperties(**gate_props)
-        for gate, props in gates.items():
-            inst = name_mapping[gate]
-            target.add_instruction(inst, props)
-        # Create measurement instructions:
-        measure_props = {}
-        for qubit, _ in enumerate(properties.qubits):
-            if filter_faulty:
-                if not properties.is_qubit_operational(qubit):
+                    in_param["error"] = param.value
+                elif param.name == "gate_length":
+                    in_param["duration"] = apply_prefix(value=param.value, unit=param.unit)
+
+            inst_prop = InstructionProperties(**in_param)
+
+            if filter_faulty and (
+                (not properties.is_gate_operational(name, gate.qubits))
+                or any(not properties.is_qubit_operational(qubit) for qubit in gate.qubits)
+            ):
+                faulty_ops.add((name, qubits))
+                try:
+                    del prop_name_map[name][qubits]
+                except KeyError:
+                    pass
+                continue
+
+            if prop_name_map[name] is None:
+                prop_name_map[name] = {}
+            prop_name_map[name][qubits] = inst_prop
+
+        # Measure instruction property is stored in qubit property
+        measure_props = []
+
+        for qubit_idx in range(configuration.num_qubits):
+            qubit_prop = properties.qubit_property(qubit_idx)
+            in_prop = {}
+            if "readout_length" in qubit_prop:
+                in_prop["duration"] = qubit_prop["readout_length"][0]
+            elif "readout_error" in qubit_prop:
+                in_prop["error"] = qubit_prop["readout_error"][0]
+            measure_props.append(InstructionProperties(**in_prop))
+
+        prop_name_map["measure"] = {}
+        for qubit, measure_prop in enumerate(measure_props):
+            if qubit in faulty_qubits:
```

Comment:
Can this check be moved bit earlier to avoid creating the InstructionProperties object that is immediately discarded. Something like

```python
        for qubit_idx in range(configuration.num_qubits):
            if qubit_idx in faulty_qubits:
                continue
            qubit_prop = properties.qubit_property(qubit_idx)    
            ...
            prop_name_map["measure"][(qubit_idx, )] = measure_prop
```
is much cleaner.

Reply:
This is indeed much clearer, I have even moved the adding of instruction properties to `prop_name_map` into this loop only to save on one extra loop :)